stable_baselines/common/policies.py

```diff
@@ -97,32 +97,68 @@ class BasePolicy(ABC):
     :param ac_space: (Gym Space) The action space of the environment
     :param n_env: (int) The number of environments to run
     :param n_steps: (int) The number of steps to run for each environment
-    :param n_batch: (int) The number of batch to run (n_envs * n_steps)
+    :param n_batch: (int) The number of batches to run (n_envs * n_steps)
     :param reuse: (bool) If the policy is reusable or not
     :param scale: (bool) whether or not to scale the input
     :param obs_phs: (TensorFlow Tensor, TensorFlow Tensor) a tuple containing an override for observation placeholder
         and the processed observation placeholder respectivly
     :param add_action_ph: (bool) whether or not to create an action placeholder
     """
 
+    stateful = False
+
     def __init__(self, sess, ob_space, ac_space, n_env, n_steps, n_batch, reuse=False, scale=False,
                  obs_phs=None, add_action_ph=False):
         self.n_env = n_env
         self.n_steps = n_steps
         with tf.variable_scope("input", reuse=False):
             if obs_phs is None:
-                self.obs_ph, self.processed_obs = observation_input(ob_space, n_batch, scale=scale)
+                self._obs_ph, self._processed_obs = observation_input(ob_space, n_batch, scale=scale)
             else:
-                self.obs_ph, self.processed_obs = obs_phs
+                self._obs_ph, self._processed_obs = obs_phs
 
-            self.action_ph = None
+            self._action_ph = None
             if add_action_ph:
-                self.action_ph = tf.placeholder(dtype=ac_space.dtype, shape=(None,) + ac_space.shape, name="action_ph")
+                self._action_ph = tf.placeholder(dtype=ac_space.dtype, shape=(n_batch,) + ac_space.shape,
+                                                 name="action_ph")
         self.sess = sess
         self.reuse = reuse
         self.ob_space = ob_space
         self.ac_space = ac_space
 
+    @property
+    def is_discrete(self):
+        return isinstance(self.ac_space, Discrete)
+
+    @property
+    def initial_state(self):
+        """
+        The initial state of the policy.
+
+        :return For stateless policies, None. For a stateful policy, a NumPy array of
+        shape (self.n_env, ) + state_shape.
+        """
+        assert not self.stateful
+        return None
+
+    @property
+    def obs_ph(self):
+        """tf.Tensor: placeholder for observations, shape (self.n_batch, ) + self.ob_space.shape."""
```

Comment:
Does this work for the automatic documentation?

Reply:
It's how the [Sphinx docs](https://www.sphinx-doc.org/en/master/usage/extensions/example_google.html?highlight=%40property) suggest documenting properties. The alternative is I could add a `:return` prefix which would turn them into zero-argument functions in the docs. Here's an example of both outputs (`neglogp` is current, `pdtype` is with `:return`). Let me know if you have a preference.

![image](https://user-images.githubusercontent.com/1547683/55364057-035f9700-5494-11e9-962b-ef465c1ec53d.png)

Reply:
I vote for your original solution.

Reply:
Yes, if it works and if it is suggested by the Sphinx doc, I don't see any pb then.